src/spyglass/utils/dj_mixin.py

```diff
@@ -133,58 +153,146 @@ def _delete_deps(self) -> list:
         return self._delete_dependencies
 
     @property
-    def _merge_del_func(self) -> callable:
-        """Callable: delete_downstream_merge function.
+    def _test_mode(self) -> bool:
+        """Return True if test mode is enabled."""
+        if not self._test_mode_cache:
+            from spyglass.settings import test_mode
+
+            self._test_mode_cache = test_mode
+        return self._test_mode_cache
 
-        Used to delay import of func until needed, avoiding circular imports.
+    @property
+    def _merge_tables(self) -> Dict[str, dj.FreeTable]:
+        """Dict of merge tables downstream of self.
+
+        Cache of items in parents of self.descendants(as_objects=True) that
+        have a merge primary key.
         """
-        if not self._merge_delete_func:
-            from spyglass.utils.dj_merge_tables import (  # noqa F401
-                delete_downstream_merge,
-            )
+        if self._merge_table_cache:
+            return self._merge_table_cache
+
+        def has_merge_pk(table):
+            return MERGE_PK in table.heading.names
+
+        self.connection.dependencies.load()
+        for desc in self.descendants(as_objects=True):
+            if not has_merge_pk(desc):
+                continue
+            if not (master_name := get_master(desc.full_table_name)):
+                continue
+            master = dj.FreeTable(self.connection, master_name)
+            if has_merge_pk(master):
+                self._merge_table_cache[master_name] = master
+        logger.info(
+            f"Building merge cache for {self.table_name}.\n\t"
+            + f"Found {len(self._merge_table_cache)} downstream merge tables"
+        )
+
+        return self._merge_table_cache
 
-            self._merge_delete_func = delete_downstream_merge
-        return self._merge_delete_func
+    @property
```

Comment:
Have you considered using [cached_property](https://docs.python.org/3/library/functools.html#functools.cached_property)?

Reply:
I did not, no. Thanks! That will save me some boiler-plate